src/common_utils/tests/test_table_filter.py

```diff
@@ -0,0 +1,183 @@
+import json
+import os
+
+import pytest
+from google.cloud.exceptions import NotFound
+
+from common_utils import table_filter
+
+MOCK_CONFIGS_DIR_NAME = "mock"
+f = open(
+    os.path.join(
+        os.path.dirname(__file__),
+        MOCK_CONFIGS_DIR_NAME,
+        "sampleconfig_for_ddl_sql_teradata.json",
+    )
+)
+f2 = open(
+    os.path.join(
+        os.path.dirname(__file__),
+        MOCK_CONFIGS_DIR_NAME,
+        "sample_config_for_sql.json",
+    ),
+    "r",
+)
+
+# Define input and expected outputs that covers:
+#     1: File Status: Success
+#     2: File Status: Fail with table already exists
+#     3: File Status: Fail with NotFound
+files_mapping = [
+    (
+        [
+            {
+                "unique_id": "unique_id_to_filter_bq_result_table",
+                "sql_file_name": "sample_sql1",
+                "job_id": "sample_job_id1",
+                "status": "success",
+                "error_details": "",
+                "execution_start_time": "20230310",
+                "gcs_source_path": "gs://edw-ams-translation-dmlcustomer/input/dml",
```

Comment:
As mentioned in another file - if this is a real bucket, could be good to put it in a variable, eg. `"gcs_source_path": f"{DMT_TEST_INPUT_BUCKET}/input/dml"`.

Reply:
Resolved: will open follow-up issue.